Engine.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class Engine {
+private:
+    int power_{};
+
+public:
+    Engine(int power);
+    int getPower() const { return power_; }
+    ~Engine() {};
+};
```

Comment:
I am not sure about Engine class here, can generate problems and force us to have additional logic for Engine checking

Reply:
Yeah this is a bit of overkill, but turned out there are not that many problems so I will left this